File: javapatterns/src/main/java/org/iut/nantes/strategy/crypto/SHA256.java
package org.iut.nantes.strategy.crypto;

import javax.xml.bind.DatatypeConverter;
import java.security.*;
import java.math.*;

public class SHA256 implements CrypteStrategy
{
  private MessageDigest m;

  public SHA256() throws Exception
  {
    m = MessageDigest.getInstance( "SHA-256" );
  }

  public String encode(String s)
  {
    return DatatypeConverter.printHexBinary( m.digest( s.getBytes() ) );
  }
}

File: src/fr/tblf/patterns/strategy/crypto/comportements/SHA256.kt
package fr.tblf.patterns.strategy.crypto.comportements

import java.security.MessageDigest

class SHA256: CryptoStrat {
    private val messageDigest: MessageDigest = MessageDigest.getInstance("SHA-256")

    override fun encrypt(message: String) {
        println(String(messageDigest.digest(message.toByteArray())))
    }
}